libs/btp/search-field/src/search-field.component.ts

```diff
@@ -0,0 +1,115 @@
+import { NgIf } from '@angular/common';
+import {
+    AfterViewInit,
+    Component,
+    DestroyRef,
+    ElementRef,
+    EventEmitter,
+    inject,
+    Input,
+    Output,
+    ViewChild
+} from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { FormsModule } from '@angular/forms';
+import { CvaControl, CvaDirective } from '@fundamental-ngx/cdk/forms';
+import { HasElementRef } from '@fundamental-ngx/cdk/utils';
+import { IconComponent } from '@fundamental-ngx/core/icon';
+import { FdTranslatePipe } from '@fundamental-ngx/i18n';
+import { of, tap } from 'rxjs';
+
+@Component({
+    selector: 'fdb-search-field',
+    template: `
```

Comment:
that's a pretty big template to be in the ts file. In my opinion it should be in a separate file. Our agreement before was that everything with more than 6 lines goes to an html file.